core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -369,4 +369,18 @@ default ClientBuilderParams validateParams(ClientBuilderParams params) {
         }
         return params;
     }
+
+    /**
+     * Registers a JVM shutdown hook that closes this {@link ClientFactory} when the current JVM terminates.
+     */
+    default CompletableFuture<Void> closeOnShutdown() {
```

Comment:
How about renaming it to `closeOnJvmShutdown`?
I think the word shutdown is also used for the server so it might be confusing. 🤔